crates/router/src/core/payment_methods/cards.rs

```diff
@@ -2193,3 +2237,42 @@ pub async fn delete_payment_method(
         },
     ))
 }
+
+pub async fn create_encrypted_payment_method_data(
+    state: &routes::AppState,
+    merchant_account: &domain::MerchantAccount,
+    pm_data: Option<PaymentMethodsData>,
+) -> Option<Encryption> {
+    let db = &*state.store;
+    let key_store = db
+        .get_merchant_key_store_by_merchant_id(
+            &merchant_account.merchant_id,
+            &db.get_master_key().to_vec().into(),
+        )
+        .await
+        .ok();
+    let key = key_store.as_ref().map(|k| k.key.get_inner().peek());
+
+    let pm_data_encrypted: Option<Encryption> = if let Some(k) = key {
+        pm_data
+            .as_ref()
+            .map(utils::Encode::<PaymentMethodsData>::encode_to_value)
+            .transpose()
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Unable to convert payment method data to a value")
+            .ok()
+            .flatten()
+            .map(masking::Secret::<_, masking::WithType>::new)
+            .async_lift(|inner| encrypt_optional(inner, k))
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
```

Comment:
encrypt failed won't be InternalServerError, there will be errors corresponds to encryption failure.

Reply:
done